src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,202 @@
+* A flag indicating that the output is not a point centered value.
+* A flag that the output is a scalar.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 13-18
+
+The ``modify_contract`` method can be used to request special information for the expression from VisIt_.
+In this case it is a no-op.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 19-20
+
+The ``derive_variable`` method performs the real work of the expression.
+It is passed the input vtkDataSet and the domain_id.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 21-23
+
+The following lines get the vtkDataArray for the cell values and the number of cells.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 24-25
+
+The following lines set the output vtkDataArray to be an array of floats with 1 component and ncells.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 26-28
+
+Now we loop over the cells, setting the output value for each cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 29
+
+The following lines get the current cell and the number of edges in the cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 30-31
+
+Now we loop over the edges, calculating the sum of the lengths of the edges.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 32-34
+
+We calculate the length of the edge from the 3D coordinates of the end points of the edge, which we add to the sum.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 35-42
+
+Once we have summed the lengths of the edges we multiply the sum by the cell value and set it in the result.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 43-45
+
+Once we have finished processing all the cells, we return the vtkDataArray.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 46
+
+Using VTK in Python
+-------------------
+
+The VTK Python interface mirrors the C++ interface.
+
+To find out information on a particular VTK class, type the name of the class in your favorite search engine.
+
+Here are links to some VTK classes that will be of most use to you.
+
+* `vtkCell <https://vtk.org/doc/nightly/html/classvtkCell.html>`_
+* `vtkDataArray <https://vtk.org/doc/nightly/html/classvtkDataArray.html>`_
+* `vtkDataSet <https://vtk.org/doc/nightly/html/classvtkDataSet.html>`_
+* `vtkDoubleArray <https://vtk.org/doc/nightly/html/classvtkDoubleArray.html>`_
+* `vtkFloatArray <https://vtk.org/doc/nightly/html/classvtkFloatArray.html>`_
+
+Using a Python Expression with the CLI
+--------------------------------------
+
+The Python expression we just created can also be used with the CLI.
+
+We'll start by saving the script we just created.
+
+1. Click *Save script* and save the script with the name "MyExpression.py".
```

Comment:
There are perhaps multiple cases of use of `"some file or variable name"` which should probably be in code typeset instead of double quotes as in 

```suggestion
1. Click *Save script* and save the script with the name ``MyExpression.py``.
```